torch/_inductor/codecache.py

```diff
@@ -772,6 +772,7 @@ def _lookup_graph(
         assert shape_env is not None
 
         symints = FxGraphCache._filter_backed_symints(example_inputs)
+        shape_env.drop_non_input_symint_guards(symints)
```

Comment:
@ezyang can you verify its safe to manipulate the existing shape env? I high-level understanding from @bdhirsh is that dynamo will still evaluate these guards after inductor is finished doing its thing. If it is safe to remove this category of guards, why don't we remove them unconditionally before we start inducting rather than putting the manipulation in the cache path?

Reply:
No, you definitely cannot start dropping guards from the shape env lol

Reply:
Fixed this on a previous version of the PR